include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -1041,15 +1420,24 @@ class __brick_set_op
     }
 };
 
-template <typename _ExecutionPolicy, typename _DiffType>
+// TODO: The implementation of shift left is reliant on exactly n (shift factor)
+// work items being launched by the parallel_for pattern, so it cannot be vectorized
+// or process multiple iterations per work items as is. For now, we must ensure that our
+// small submitter is launched in the SYCL backend's __parallel_for
+template <typename _ExecutionPolicy, typename _DiffType, typename _Range>
```

Comment:
I'm pretty sure this is vectorizable.  The parallelism is "within" the shift, so if we have a shift of 2048, there are 2048 parallel lines of execution.  Each iteration of the for loop contains a contiguous "next" segment of 2048 which is loaded, and then written to the contiguous "current" segment of 2048.  Seems like just a loop with a load followed by a store should vectorize it just fine.

No gathers / scatters should be necessary.  Of course, without a very large shift, the utility of this case is not great, but vectorization should be better than scalar all the same.

Reply:
Thanks for convincing me of this. The vectorization path has been added and speedups are in line with previous data I have shared.

Reply:
great!  Left a couple minor comments on the impl, but I'll resolve this one.